include/net/nrf_cloud_os.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef NRF_CLOUD_OS_H__
+#define NRF_CLOUD_OS_H__
+
+/** @defgroup nrf_cloud_os nRF Cloud OS specifics
+ * @{
+ */
+
+#include <stdlib.h>
+#include <zephyr/kernel.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief Custom OS memory hooks for nRF Cloud lib */
+struct nrf_cloud_os_mem_hooks {
+	void *(*malloc_fn)(size_t size);
+	void *(*calloc_fn)(size_t count, size_t size);
+	void (*free_fn)(void *ptr);
+};
+
+/**
+ * @brief As a default nRF Cloud lib uses OS kernel heap memory
+ *        (i.e. k_ prepending OS functions), this can be used to override those
+ *        (e.g. to be using OS system heap, i.e. standard C calloc/malloc/free).
+ *
+ * @note This API must be called prior to using nRF Cloud, even before
```

Comment:
you could add a `struct nrf_cloud_os_mem_hooks *hooks` to `struct nrf_cloud_init_param` so the hooks could be set at init also

Reply:
OK, added a possibility to set those hooks also from nrf_cloud_init().